### PR TITLE
Implement profanity filter Fixes #162

### LINKED ISSUE #162: Filter some dirty words

### PR DESCRIPTION
**Pull request by Jaapp-**
Create a profanity filter that will match words and replace profanity with ****'s.

### CODE REVIEW

File: src/main/resources/profanity/nl.txt [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+sufferdg
+tapijtnek
+teefg
+temeier
+teringlijer
+toeter
+tongzoeng
+triootjeg
+trottoirteef
+utrecht
```
Comment (Azkle) [inline src/main/resources/profanity/nl.txt]: This is a normal word.

File: src/main/java/me/moodcat/backend/rooms/RoomInstance.java [hunk 2]
```diff
@@ -58,6 +62,11 @@
     private final UnitOfWorkSchedulingService unitOfWorkSchedulingService;
 
     /**
+     * The profanity checker to filter out 'bad' chatmessages.
+     */
+    private final ProfanityChecker profanityChecker = new ProfanityChecker();
```
Comment (JWGmeligMeyling) [inline src/main/java/me/moodcat/backend/rooms/RoomInstance.java]: Setting this through the constructor and instantiating through Guice will technically make this instance singleton, which saves memory. It also makes the implementation more testable/mockable. So I agree with Tim on this.

**Comment by JWGmeligMeyling** [comment]
Good! :+1:

**Comment by Azkle** [comment]
LGTM, only the Dutch list is not that great.

**Comment by Azkle** [comment]
Mergeable! Looks good.

**Pull request merged**